skrf/circuit.py

```diff
@@ -1540,17 +1616,24 @@ def reduce_circuit(connections: list[list[tuple[Network, int]]],
 
     Parameters
     ----------
-    connections : list.
+    connections : list[list[tuple[Network, int]]].
             The connection list to reduce.
     check_duplication : bool, optional.
             If True, check if the connections have duplicate names. Default is True.
     split_ground : bool, optional.
-            If True, split the global ground connection to independant ground connections. Default is False.
-    max_nports : int
+            If True, split the global ground connection to independant ground connections. Default is True.
+    split_multi : bool, optional.
+            If True, use a splitter to handle connections involving more than two components. This approach
+            increases the computational load for individual computations. However, it proves advantageous for
+            batch processing by enabling a more comprehensive reduction of circuits, leading to more efficiency
+            in batch computations. Default is False.
+    max_nports : int, optional.
             The maximum number of ports of a Network that can be reduced in circuit. If a Network in the
             circuit has a number of ports (nports), using the Network.connect() method to reduce the circuit's
             dimensions becomes less efficient compared to directly calculating it with Circuit.s_external.
             This value depends on the performance of the computer and the scale of the circuit. Default is 20.
+    dynamic_networks : tuple[Network, ...], optional.
```

Comment:
```suggestion
    dynamic_networks : list of tuples (Network, port), optional.
```
or something similar?

Reply:
Since we aim to update all specified `dynamic_networks` in the `circuit object`, the input parameters neither need nor should include any port index information. Doing so would cause confusion and imply that only Networks with a **completely matching `Network and port index`** will be updated.

To meeting the typing hinting requirements, using `tuple[Network, ...]` is great choice, as it allows for a tuple of `Networks` with any length.

> To denote a tuple which could be of **any length**, and in which all elements are of the same type `T`, use tuple`[T, ...]`. To denote an empty tuple, use tuple`[()]`. Using plain tuple as an annotation is equivalent to using tuple`[Any, ...]`:

Reply:
Ok, clear.

Reply:
I think `Sequence` would be a proper fit, as both `tuple` and `list` are valid `Sequence`.
https://stackoverflow.com/questions/42486006/what-type-hint-contains-both-list-and-tuple

Consider that only `typing.Sequence` is deprecated, `collections.abc.Sequence` is still fine.

Reply:
Good suggestion! The update has been implemented as advised.
Given that `collections.abc.Sequence` is rarely utilized in `scikit-rf`, I opted for the deprecated `typing.Sequence` to ensure consistency across the codebase.